cpp/cmake_modules/FindPythonLibsNew.cmake

```diff
@@ -234,11 +234,7 @@ FUNCTION(PYTHON_ADD_MODULE _NAME )
       SET_TARGET_PROPERTIES(${_NAME} PROPERTIES LINK_FLAGS
                           "-undefined dynamic_lookup")
     ELSE()
```

Comment:
Got it, very helpful. I'll restore the prior logic for Linux only (MSVC has to link to libpython) and then merge